python/composio/tools/env/docker/shell.py

```diff
@@ -160,6 +160,11 @@ def _communicate_with_handling(self, cmd: str, error: str) -> str:
             raise RuntimeError(f"{cmd}: {error}: {output}")
         return output["stdout"]
 
+    def _get_exit_code(self) -> int:
+        """Get exit code of the last process."""
+        self._write("echo $#")
```

Comment:
this should be "echo $?" not "$#", $# gives number of output printed, #? gives correct exit_status of last command executed